compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/CheckExpressionTest.java

```diff
@@ -36,6 +36,41 @@ public void testCheckExprPrecedence() {
         testFile("check-expr/check_expr_source_02.bal", "check-expr/check_expr_assert_02.json");
     }
 
+    @Test
+    public void testCheckExprWithOnConflictClause() {
+        testFile("check-expr/check_expr_source_05.bal", "check-expr/check_expr_assert_05.json");
+    }
+
+    @Test
+    public void testSimpleCheckOnFail() {
+        testFile("check-expr/on_fail_check_source_01.bal",  "check-expr/on_fail_check_assert_01.json");
+    }
+
+    @Test
+    public void testOnFailCheckpanic() {
+        testFile("check-expr/on_fail_check_source_04.bal", "check-expr/on_fail_check_assert_04.json");
+    }
+
+    @Test
+    public void testOnFailCheckWithReturn() {
+        testFile("check-expr/on_fail_check_source_05.bal", "check-expr/on_fail_check_assert_05.json");
+    }
+
+    @Test
+    public void testOnFailCheckpanicWithReturn() {
+        testFile("check-expr/on_fail_check_source_06.bal", "check-expr/on_fail_check_assert_06.json");
+    }
+
+    @Test
+    public void testOnFailCheckNextToAnOnConflictClause() {
+        testFile("check-expr/on_fail_check_source_08.bal", "check-expr/on_fail_check_assert_08.json");
+    }
+
+    @Test
+    public void testOnFailWithJoin() {
```

Comment:
```suggestion
    public void testCheckExprWithJoinClause() {
```

Also, move it next to the `testCheckExprWithOnConflictClause()` as they are similar tests.